onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -0,0 +1,251 @@
+    if (p_sub_node_dup != nullptr) {
+      Node& sub_node_dup = *graph.GetNode(p_sub_node_dup->Index());
+      if (!graph_utils::IsSupportedOptypeVersionAndDomain(sub_node_dup, "Sub", {7}) ||
+          sub_node_dup.GetExecutionProviderType() != reduce_mean_node.GetExecutionProviderType() ||
+          sub_node_dup.GetOutputEdgesCount() != 1 ||
+          !IsSupportedDataType(sub_node_dup)) {
+        continue;
+      }
+      nodes_to_remove.push_back(sub_node_dup);
+      // Find Div node after the duplicated sub node if it's not found after the first sub node. 
+      if (p_div == nullptr) {
+        p_div = graph_utils::FirstChildByType(sub_node_dup, "Div");
+      }
+    }
+
+    if (p_div == nullptr) {
+      continue;
+    }
+    Node& div_node = *graph.GetNode(p_div->Index());
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(div_node, "Div", {7}) ||
+        div_node.GetExecutionProviderType() != reduce_mean_node.GetExecutionProviderType() ||
+        div_node.GetOutputEdgesCount() != 1 ||
+        !IsSupportedDataType(div_node)) {
+      continue;
+    }
+    nodes_to_remove.push_back(div_node);
+
+    // Traceback the div node to find sqrt --> div
+    const Node* p_sqrt = graph_utils::FirstParentByType(div_node, "Sqrt");
+    if (p_sqrt == nullptr) {
+      continue;
+    }
+    Node& sqrt_node = *graph.GetNode(p_sqrt->Index());
+
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(sqrt_node, "Sqrt", {6}) ||
+        sqrt_node.GetExecutionProviderType() != reduce_mean_node.GetExecutionProviderType() ||
+        sqrt_node.GetOutputEdgesCount() != 1 ||
+        !IsSupportedDataType(sqrt_node)) {
+      continue;
+    }
+    nodes_to_remove.push_back(sqrt_node);
+
+    // Traceback the sqrt node to find add --> sqrt
+    Node& add2_node = *graph.GetNode(sqrt_node.InputNodesBegin()->Index());
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(add2_node, "Add", {7}) ||
+        add2_node.GetExecutionProviderType() != reduce_mean_node.GetExecutionProviderType() ||
+        add2_node.GetOutputEdgesCount() != 1 ||
+        !IsSupportedDataType(add2_node)) {
+      continue;
+    }
+    nodes_to_remove.push_back(add2_node);
+
+    // Traceback the add node to find reduceMean --> add
+    const Node* p_reduce_mean2 = nullptr;
+
+    p_reduce_mean2 = graph_utils::FirstParentByType(add2_node, "ReduceMean");
+    Node& reduce_mean2_node = *graph.GetNode(p_reduce_mean2->Index());
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(reduce_mean2_node, "ReduceMean", {1}) ||
+        reduce_mean2_node.GetExecutionProviderType() != reduce_mean_node.GetExecutionProviderType() ||
+        reduce_mean2_node.GetOutputEdgesCount() != 1 ||
+        !IsSupportedDataType(reduce_mean2_node)) {
+      continue;
+    }
+    nodes_to_remove.push_back(reduce_mean2_node);
+
+    // Traceback the reduceMean node to find pow --> reduceMean
+    Node& pow_node = *graph.GetNode(reduce_mean2_node.InputNodesBegin()->Index());
```

Comment:
>reduce_mean2_node.InputNodesBegin()->Index()); [](start = 36, length = 46)

check if the input node exist. reduce_mean2_node's input node may be null if all of its input are graph input